File: chat/uikit/ios/Classes/TencentChatUikitPlugin.swift
import Flutter
import UIKit

/// TencentChatUikitPlugin
///
/// Hosts the chat-business native handlers that previously lived in
/// AtomicXPlugin. Module/handler ownership matches the Dart-side migration:
///   audio_player, audio_recoder, file_picker, video_player, video_recorder.
///
/// 托管之前在 AtomicXPlugin 中的聊天业务原生处理程序。模块/处理程序的归属与 Dart
/// 端迁移一致：audio_player、audio_recoder、file_picker、video_player、video_recorder。
public class TencentChatUikitPlugin: NSObject, FlutterPlugin {
  private var videoRecorder: AtomicVideoRecorderPlugin?
  private var audioRecorder: AtomicAudioRecorderPlugin?
  private var audioPlayer: AtomicAudioPlayerPlugin?
  private var filePicker: AtomicFilePickerPlugin?
  private var videoPlayer: AtomicVideoPlayerPlugin?

  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(
      name: "tencent_chat_uikit",
      binaryMessenger: registrar.messenger()
    )
    let instance = TencentChatUikitPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)

    instance.videoRecorder = AtomicVideoRecorderPlugin(registrar: registrar)
    instance.audioRecorder = AtomicAudioRecorderPlugin(registrar: registrar)
    instance.audioPlayer = AtomicAudioPlayerPlugin(registrar: registrar)
    instance.filePicker = AtomicFilePickerPlugin(registrar: registrar)
    instance.videoPlayer = AtomicVideoPlayerPlugin.register(with: registrar) as? AtomicVideoPlayerPlugin
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getPlatformVersion":
      result("iOS " + UIDevice.current.systemVersion)
    default:
      result(FlutterMethodNotImplemented)
    }
  }

  deinit {
    videoRecorder?.dispose()
    audioRecorder?.dispose()
    audioPlayer?.dispose()
    filePicker?.dispose()
  }
}

File: chat/uikit/android/src/main/kotlin/com/tencent/cloud/tuikit/flutter/tuichatkit/TencentChatUikitPlugin.kt
package com.tencent.cloud.tuikit.flutter.tuichatkit

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.embedding.engine.plugins.activity.ActivityAware
import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding
import com.tencent.cloud.tuikit.flutter.tuichatkit.audioplayer.AtomicAudioPlayerPlugin
import com.tencent.cloud.tuikit.flutter.tuichatkit.audiorecorder.AtomicAudioRecorderPlugin
import com.tencent.cloud.tuikit.flutter.tuichatkit.filepicker.AtomicFilePickerPlugin
import com.tencent.cloud.tuikit.flutter.tuichatkit.videoplayer.AtomicVideoPlayerPlugin
import com.tencent.cloud.tuikit.flutter.tuichatkit.videorecorder.AtomicVideoRecorderPlugin

/** TencentChatUikitPlugin
 *
 * Hosts the chat-business native handlers that were previously bundled inside
 * AtomicXPlugin. Module/handler ownership matches the Dart-side migration:
 *   audio_player, audio_recoder, file_picker, video_player, video_recorder.
 *
 * 托管之前包含在 AtomicXPlugin 里的聊天业务原生处理器。模块/处理器的归属对应 Dart
 * 端的迁移：audio_player、audio_recorder、file_picker、video_player、video_recorder。
 */
class TencentChatUikitPlugin : FlutterPlugin, ActivityAware {
  companion object {
    private const val TAG = "TencentChatUikitPlugin"
  }

  private var audioPlayerPlugin: AtomicAudioPlayerPlugin? = null
  private var audioRecorderPlugin: AtomicAudioRecorderPlugin? = null
  private var filePickerPlugin: AtomicFilePickerPlugin? = null
  private var videoPlayerPlugin: AtomicVideoPlayerPlugin? = null
  private var videoRecorderPlugin: AtomicVideoRecorderPlugin? = null

  override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    videoRecorderPlugin = AtomicVideoRecorderPlugin(flutterPluginBinding)
    audioRecorderPlugin = AtomicAudioRecorderPlugin(flutterPluginBinding)
    audioPlayerPlugin = AtomicAudioPlayerPlugin().also {
      it.onAttachedToEngine(flutterPluginBinding)
    }
    filePickerPlugin = AtomicFilePickerPlugin().also {
      it.onAttachedToEngine(flutterPluginBinding)
    }
    videoPlayerPlugin = AtomicVideoPlayerPlugin().also {
      it.onAttachedToEngine(flutterPluginBinding)
    }
  }

  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    videoRecorderPlugin?.dispose()
    videoRecorderPlugin = null
    audioRecorderPlugin?.dispose()
    audioRecorderPlugin = null
    audioPlayerPlugin?.onDetachedFromEngine(binding)
    audioPlayerPlugin = null
    filePickerPlugin?.onDetachedFromEngine(binding)
    filePickerPlugin = null
    videoPlayerPlugin?.onDetachedFromEngine(binding)
    videoPlayerPlugin = null
  }

  override fun onAttachedToActivity(binding: ActivityPluginBinding) {
    audioRecorderPlugin?.attachToActivity(binding.activity)
  }

  override fun onDetachedFromActivityForConfigChanges() {}

  override fun onReattachedToActivityForConfigChanges(binding: ActivityPluginBinding) {}

  override fun onDetachedFromActivity() {
    audioRecorderPlugin?.detachFromActivity()
  }
}
